syntax = "proto3";

package kv;

message SetRequest {
    string key = 1;
    string value = 2;
}

message SetResponse {
    bool ok = 1;
}

message UnsetRequest {
    string key = 1;
}

message UnsetResponse { 
    bool ok = 1;
}

message IncrRequest {
    string key = 1;
    int64 incr = 2;
}

message IncrResponse {
    bool ok = 1;
    int64 new_val = 2;
}

message GetRequest {
    string key = 1;
}

message GetResponse {
    string value = 1;
}

service KVStore {
    rpc Set(SetRequest) returns (SetResponse) {}
    rpc Get(GetRequest) returns (GetResponse) {}
    rpc Incr(IncrRequest) returns (IncrResponse) {}
    //rpc Monitor(GetRequest) returns (stream GetResponse) {}
}